src/components/document/document-workspace.tsx

```diff
@@ -163,6 +166,28 @@ export class DocumentWorkspaceComponent extends BaseComponent<IProps, {}> {
     this.stores.ui.contractAll();
   }
 
+  private handleNewDocument = (document: DocumentModelType) => {
+    const docType = document.isLearningLog ? LearningLogDocument : PersonalDocument;
+    const docTypeString = document.isLearningLog ? "Learning Log" : "Personal Document";
+    const { appConfig: { defaultDocumentTitle } } = this.stores;
+    const nextTitle = this.stores.documents.getNextOtherDocumentTitle(this.stores.user, docType, defaultDocumentTitle);
+    this.stores.ui.prompt(`Name your new ${docTypeString}:`, `${nextTitle}`, `Create ${docTypeString}`)
+      .then((title: string) => {
+        this.handleNewDocumentOpen(docType, title)
+        .catch(this.stores.ui.setError);
+      });
+  }
+
+  private handleNewDocumentOpen = async (type: OtherDocumentType, title: string) => {
+    const { appConfig, db, ui: { problemWorkspace } } = this.stores;
+    const content = (type === PersonalDocument) && appConfig.defaultDocumentTemplate
+                      ? appConfig.defaultDocumentContent : undefined;
+    const newDocument = await db.createOtherDocument(type, {title, content});
+    if (newDocument) {
+      problemWorkspace.setAvailableDocument(newDocument);
+    }
+  }
+
```

Comment:
@llzes is the new doc button supposed to appear in the side panel?  We might need to remove it in the Dataflow app config.

Reply:
In `app-config.json` for Dataflow, the following line needs to be removed:

[app-config.json](https://github.com/concord-consortium/collaborative-learning/blob/merge-master/src/dataflow/app-config.json#L41)
```
"addDocument": true
```